Resource/PYTHON/multilateration/multilateration.py

```diff
@@ -0,0 +1,287 @@
+                2,
+            )
+            + math.pow(
+                param.HydrophoneDetails.STERN_HYD_Z
+                - param.HydrophoneDetails.STARBOARD_HYD_Z,
+                2,
+            )
+        )
+    )
+
+    # Finding the maximum distance
+    max_hydrophone_distance = max(
+        dist_port_starboard, max(dist_starboard_stern, dist_port_stern)
+    )
+
+    # Calculating max time allowed over that distance
+    max_time_diff = (1 + param.SystemMargins.MARGIN_TIME_EPSILON) * (
+        max_hydrophone_distance / param.PhysicalConstants.SOUND_SPEED
+    )
+
+    # Returning if both variables have been set correctly
+    return check_initialized_globals()
+
+
+def initialize_A_matrix():  # return type Matrix_2_3_f
+    A = np.zeros((2, 3))
+    return A
+
+
+def initialize_B_vector():
+    B = np.zeros((2, 1))
+    return B
+
+
+# Functions to check if signals/data are valid
+
+
+def check_initialized_globals():  # return type uint8_t
+    return max_hydrophone_distance != -1 and max_time_diff != 1
+
+
+def check_valid_time(time_lhs: np.uint32, time_rhs: np.uint32):  # return type uint8_t
+
+    # Calculating the time-difference and checking if it exceeds the maximum
+    # allowed time for a valid signal
+
+    time_diff = time_lhs - time_rhs  # type int32_t
+    return (abs(time_diff) * param.DSPConstants.SAMPLE_TIME) > max_time_diff
+
+
+def check_valid_time(time_diff: np.uint32):  # return type uint8_t
+
+    # Checking if the time_diff exceeds the maximum allowed time for a valid signal
+
+    return (abs(time_diff) * param.DSPConstants.SPSAMPLE_TIME) > max_time_diff
+
+
+def check_valid_signals(p_lag_array: np.array, bool_time_error: np.uint32):
+    """
+    arguments:
+      in the orginal code is was written as such: uint32_t* p_lag_array[NUM_HYDROPHONES],uint8_t& bool_time_error)
+
+    returns:
+      type uint8
+    """
+
+    # Evaluating if the signals are valid in time
+    if (
+        check_valid_time(p_lag_array[0])
+        or check_valid_time(p_lag_array[1])
+        or check_valid_time(p_lag_array[2])
+    ):
+        bool_time_error = 1
+        # Possible to add other tests here in the future
+
+        # Returning the test-results
+    return not (bool_time_error)
+
+
+# Functions for trilateration based on TDOA
+
+
+def trilaterate_pinger_position(
+    p_lag_array,
+):  # obs: originaly x_estimate og y_estimate were taken in as arguments
+    """
+    arg:
+    A is a Matrix_2_3_f
+    B is a Vector_2_1_f
+    p_lag_array has length NUM_HYDROPHONES
+    x_estimate and y_estimate is a float32
+
+    returns:
+
+    """
+    # Calculating TDOA [s] and creating an array to hold the data
+    TDOA_port_starboard = param.DSPConstants.SAMPLE_TIME * p_lag_array[0]
+    TDOA_port_stern = param.DSPConstants.SAMPLE_TIME * p_lag_array[1]
+    TDOA_starboard_stern = param.DSPConstants.SAMPLE_TIME * p_lag_array[2]
+
+    TDOA_array = [TDOA_port_starboard, TDOA_port_stern, TDOA_starboard_stern]
```

Comment:
Even if this is an acronym, please don't capitalize variables. [See pep8](https://www.python.org/dev/peps/pep-0008/#toc-entry-32). I'm somewhat okay with it if it's for matrices, but not this.